nipype/pipeline/engine/nodes.py

```diff
@@ -610,177 +539,131 @@ def _load_results(self, cwd):
+        except Exception as msg:
+            result.runtime.stderr = '%s\n\n%s' % (
+                getattr(result.runtime, 'stderr', ''), msg)
+            _save_resultfile(result, outdir, self.name)
+            raise
+
+        dirs2keep = None
+        if isinstance(self, MapNode):
+            dirs2keep = [op.join(outdir, 'mapflow')]
+
+        result.outputs = clean_working_directory(
+            result.outputs, outdir,
+            self._interface.inputs,
+            self.needed_outputs,
+            self.config,
+            dirs2keep=dirs2keep
+        )
+        _save_resultfile(result, outdir, self.name)
+
+        return result
 
-    def _copyfiles_to_wd(self, outdir, execute, linksonly=False):
+    def _copyfiles_to_wd(self, execute=True, linksonly=False):
         """ copy files over and change the inputs"""
-        if hasattr(self._interface, '_get_filecopy_info'):
-            logger.debug('copying files to wd [execute=%s, linksonly=%s]',
-                         str(execute), str(linksonly))
-            if execute and linksonly:
-                olddir = outdir
-                outdir = op.join(outdir, '_tempinput')
-                os.makedirs(outdir)
-            for info in self._interface._get_filecopy_info():
-                files = self.inputs.get().get(info['key'])
-                if not isdefined(files):
-                    continue
-                if files:
-                    infiles = filename_to_list(files)
-                    if execute:
-                        if linksonly:
-                            if not info['copy']:
-                                newfiles = copyfiles(infiles,
-                                                     [outdir],
-                                                     copy=info['copy'],
-                                                     create_new=True)
-                            else:
-                                newfiles = fnames_presuffix(infiles,
-                                                            newpath=outdir)
-                            newfiles = self._strip_temp(
-                                newfiles,
-                                op.abspath(olddir).split(op.sep)[-1])
-                        else:
-                            newfiles = copyfiles(infiles,
-                                                 [outdir],
-                                                 copy=info['copy'],
-                                                 create_new=True)
+        if not hasattr(self._interface, '_get_filecopy_info'):
+            # Nothing to be done
+            return
+
+        logger.debug('copying files to wd [execute=%s, linksonly=%s]',
+                     execute, linksonly)
+
+        outdir = self.output_dir()
+        if execute and linksonly:
+            olddir = outdir
+            outdir = op.join(outdir, '_tempinput')
+            makedirs(outdir, exist_ok=True)
+
+        for info in self.interface._get_filecopy_info():
```

Comment:
`self._interface` and `self.interface` should be equal - but as an added layer of security wouldn't it be better to keep the usage consistent?